### PR TITLE
ARTS 331 - UI Storage Account + Networking issue fixed

### PR DESCRIPTION
**Pull request by eladiw**
This PR:
1. Adds a storage account for the UI component
2. change subnets and vnets to fix private link
   2.1 work according to https://docs.microsoft.com/en-us/azure/architecture/example-scenario/private-web-app/private-web-app
   2.2 2 subnets: first for the private endpoints and one for the integration point
3. Enable logging for web apps

### CODE REVIEW

File: sample_trial/definition.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
-    "name": "sampletrial",
-    "id": "sample1",
+    "name": "mytrial",
```
Comment (balteravishay) [inline sample_trial/definition.json:2]: is this change intentional?

File: terraform/trial_rg/main.tf [hunk 2]
```diff
@@ -115,15 +116,31 @@ resource "random_password" "roles_sql_password" {
...
+}
+
+# Storage account for UI elemenet
+resource "azurerm_storage_account" "uistorageaccount" {
+  name                      = "sa${var.trial_name}ui${var.environment}"
+  resource_group_name       = azurerm_resource_group.trial_rg.name
+  location                  = azurerm_resource_group.trial_rg.location
+  account_kind              = "StorageV2"
+  account_tier              = "Standard"
+  account_replication_type  = "GRS"
```
Comment (balteravishay) [inline terraform/trial_rg/main.tf]: why GRS?

File: terraform/trial_rg/modules/fhir/main.tf [hunk 3]
```diff
@@ -27,17 +27,28 @@ resource "azurerm_app_service" "fhir_server" {
 
   site_config {
     linux_fx_version = "DOCKER|${var.fhir_image_name}:${var.fhir_image_tag}"
+    always_on        = true
+  }
+
+  logs {
+    http_logs {
```
Comment (balteravishay) [inline terraform/trial_rg/modules/fhir/main.tf:34]: what about application logs and failed requests? may want these as well

File: terraform/trial_rg/modules/privateendpoint/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,4 @@
+output "private_link_endpoint_ip" {
```
Comment (balteravishay) [inline terraform/trial_rg/modules/privateendpoint/outputs.tf:1]: might want to name which private link this is, since you'll end up with some.

Comment (balteravishay) [reply terraform/trial_rg/modules/privateendpoint/outputs.tf:1]: akv will need one also

File: sample_trial/definition.json (hunk 1 above)
Comment (eladiw) [reply sample_trial/definition.json:2]: yes. since others worked on this repo I didnt want to override their / mine deployment

File: terraform/trial_rg/main.tf (hunk 2 above)
Comment (eladiw) [reply terraform/trial_rg/main.tf]: Should be LRS

File: terraform/trial_rg/modules/fhir/main.tf (hunk 3 above)
Comment (eladiw) [reply terraform/trial_rg/modules/fhir/main.tf:34]: agree

File: terraform/trial_rg/modules/privateendpoint/outputs.tf (hunk 4 above)
Comment (eladiw) [reply terraform/trial_rg/modules/privateendpoint/outputs.tf:1]: this is a reusable module.
I call it twice....

File: sample_trial/definition.json (hunk 1 above)
Comment (tamirkamara) [reply sample_trial/definition.json:2]: I think we should have a task to check that the folder name is the same as the name of the trial. Otherwise, the result can be unexpected...

File: terraform/trial_rg/main.tf [hunk 5]
```diff
@@ -115,15 +116,31 @@ resource "random_password" "roles_sql_password" {
 
 # Create a roles and permissions SQL
 # Deploy a sql server and db for fhir before we create the web app
-module "fhir_sql_server" {
+module "roles_sql_server" {
```
Comment (tamirkamara) [inline terraform/trial_rg/main.tf:119]: Should we have a sql module that could be "turned" on for any service?

**Review by tamirkamara — APPROVED** [review]
Approved, but added 2 comments about things I think we should do and open tasks for.

**Review by balteravishay — APPROVED** [review]

File: terraform/trial_rg/main.tf (hunk 5 above)
Comment (shajidissa) [inline terraform/trial_rg/main.tf:119]: why has fhir_sql_server been renamed to roles_sql_server ? is there a distinction between the FHIR and data sql server instances ?

**Review by shajidissa — APPROVED** [review]

File: terraform/trial_rg/main.tf (hunk 5 above)
Comment (eladiw) [reply terraform/trial_rg/main.tf:119]: I have created 2 sql servers. one for fhir and one for roles

Comment (eladiw) [reply terraform/trial_rg/main.tf:119]: currently two sql servers;
1. fhir
2. roles.


the second can have multiple dbs if we'd like so all services can access. wdyt?
